Lit: Initial lit2 support

brion-fuller opened the pull request:
Issue: #14455 

## What I did

Initial start to setting up lit framework

Commit:
feat(lit): add initial kitchen sink for lit

Commit:
feat(lit): add lit package, setup eslint configs, initial render works

gaetanmaisse committed:
Merge branch 'next' into @storybook/lit

gaetanmaisse committed:
refactor(lit): activate TS strict mode and fix errors

gaetanmaisse committed:
docs(lit): remove ref to web components and add a big warning

gaetanmaisse committed:
example(lit): remove addon knob as it will be deprecated in favor of addon control

gaetanmaisse committed:
fixup! example(lit): remove addon knob as it will be deprecated in favor of addon control

gaetanmaisse committed:
refactor(docs): remove babel-loader peer dep

`babel-loader` is required in the context of `builder-webpack4` so there is no no need to have it as a dep.

gaetanmaisse committed:
refactor(builders): require react in the SB app context as it is providing it

shilman commented:
@gaetanmaisse these last set of changes (addon-docs/babel-loader stuff) look good, but probably should be in their own PR!

gaetanmaisse commented:
Yep @shilman, I will open a PR with these and maybe some other related stuff ;)

gaetanmaisse committed:
example(lit): move the example out the Yarn workspaces of the monorepo

The goal is to avoid adding dependencies related to the example in the SB codebase workpaces.
Considering this example as a stand-alone workspace make it works in a more realistic way.

Working on things in `@storybook/lit` and want to test them in lit-kitchen-sink?
Run:
 - `yarn build lit --watch` in the monorepo root folder
 - `yarn install` then `yarn storybook` in lit-kitchen-sink folder

You're good to go, both changes in the example code or in `@storybook/lit` will be reflected in the UI.

shilman commented on examples/lit-kitchen-sink/README.md:
```diff
@@ -3,3 +3,5 @@
 
 Run `yarn install` to sync Storybook module with the source code and run `yarn storybook` to start the Storybook.
+
+When developing things in `@storybook/lit` you can run `yarn build lit --watch` in the root folder of the monorepo and both changes in the example code or in `@storybook/lit` will be reflected in the UI.
```
NICE 💯

gaetanmaisse committed:
Merge branch 'next' into @storybook/lit

gaetanmaisse committed:
example(lit): update to lit rc and fix stories

gaetanmaisse committed:
Merge branch 'next' into @storybook/lit

gaetanmaisse committed:
Revert "refactor(builders): require react in the SB app context as it is providing it"

This reverts commit a652a013 as it needs some extra work that should be done in a separated PR

gaetanmaisse committed:
example(lit): add stories for Addon Actions

gaetanmaisse committed:
example(lit): add stories for Addon Background

gaetanmaisse committed:
example(lit): add stories for Addon Viewport

gaetanmaisse committed:
example(lit): add stories for Addon Toolbars

gaetanmaisse committed:
feat(lit): bootstrap support of Lit in Addon Docs

gaetanmaisse committed:
example(lit): add basic stories for Addon Docs

gaetanmaisse committed:
example(lit): add stories for Addon Controls and improve "demo button"

gaetanmaisse committed:
Merge branch 'next' into @storybook/lit

gaetanmaisse committed:
refactor(docs): extract prepareForInline and fix to TS errors

gaetanmaisse committed:
example(lit): remove stories for deprecated features `info` and `note`

gaetanmaisse committed:
example(lit): filter `lit-kitchen-sink` from `yarn build-storybooks` command for now

As `lit-kitchen-sink` is not part of the monorepo it requires some extra work to make it build properly as we need to install deps first.

brion-fuller commented:
@gaetanmaisse Looks good to me.

shilman committed:
Merge branch 'next' into pr/14600

gaetanmaisse committed:
feat(lit): fix Deepscan warning and strengthen TypeScript compilation options

gaetanmaisse committed:
example(lit): remove unused addons from dependencies

shilman reviewed (approved):
Looking great @brion-fuller @gaetanmaisse 💯
